### DISCUSSION TITLE
How to upgrade PS 8.1.7 to 8.2.0 on PHP 8.2.26?

### DISCUSSION THREAD

**Discussion by BigOHenry** [PrestaShop/PrestaShop#37599] [Q&A] [answered]
I have PS 8.1.7 on PHP 8.2.26 and i want to upgrade to PS8.2.0. The problem is that 1click upgrade says that I have an incompatible version of PHP and that I must have 7.2.5 - 8.1.

The funny thing is that all the time I have prestashop 8.1.0 installed I have PHP8.2. What has changed? How can I upgrade?

Thank you very much, Andrew

**Comment by Hlavtox** [comment, marked as answer]
PrestaShop 8.1 is not officially compatible with PHP 8.2. Downgrade to PHP 8.1, run the upgrade.

You can use PHP 8.2-8.4 after PrestaShop 9 is released. :-)

**Reply by BigOHenry** [reply]
Thank you for response. So if its not supported, why i was able to install PS8.1 and all the updates to 8.1.7 on PHP 8.2?

**Reply by matks** [reply]
Because PHP8.1 and PHP8.2 are very similar so 80% of the software works out-of-the-box with PHP8.2 😉 

But in order to say "PS 8.2.0 is compatible with PHP8.2" we need 100%

**Reply by BigOHenry** [reply]
I know how it works, I'm a PHP developer. And that's why I'm asking why PS allowed me to install PS8.1 and other updates and suddenly not? So why didn't he prevent it from the beginning?

**Reply by matks** [reply]
I think it's simply because the autoupgrade module performs more checks than the installer

**Reply by BigOHenry** [reply]
Is there any way, to modify some files/configuration of autoupgrade to be able to skip checks and run the update?

**Reply by matks** [reply]
The php version is being checked there https://github.com/PrestaShop/autoupgrade/blob/dev/classes/UpgradeSelfCheck.php#L605

It's used here https://github.com/PrestaShop/autoupgrade/blob/dev/views/templates/block/checklist.html.twig#L23

and the logic which chooses whether or not the "Upgrade!" button is here
https://github.com/PrestaShop/autoupgrade/blob/37ed65ce28ac44a1382d947bf70071e84b7c45bc/classes/Twig/Block/UpgradeButtonBlock.php#L108

**Reply by BigOHenry** [reply]
Thank you for show me the way! All that was needed is return PHP_REQUIREMENTS_VALID here: https://github.com/PrestaShop/autoupgrade/blob/v6.2.0/classes/UpgradeSelfCheck.php#L484
